pkg/resource/deploy/plan.go

```diff
@@ -65,6 +65,7 @@ type Events interface {
 	OnResourceStepPre(step Step) (interface{}, error)
 	OnResourceStepPost(ctx interface{}, step Step, status resource.Status, err error) error
 	OnResourceOutputs(step Step) error
+	RemovePendingReplacements(res []*resource.State) error
```

Comment:
This seems a little out-of-place on the `Events` interface. Should we instead be thinking about an `OnPlanCompleted` event or something that passes some state, including the set of resources that were pending replacement but were never replaced?

Reply:
It turns out that having this be a bulk operation didn't really work in any case: it was possible for pending-delete resources to remain in the checkpoint after things _they_ depend on were deleted. I've changed the way we schedule these operations to be an internal step.